src/main/java/pingis/controllers/TaskController.java

```diff
@@ -64,35 +64,43 @@ public String task(Model model,
         model.addAttribute("taskImplementationId",
                 taskImplementationId);
         Map<String, EditorTabData> editorContents = editorService.generateEditorContents(taskImplementation);
-        model.addAttribute("testCode", editorContents.get("editor1").code);
-        model.addAttribute("implementationCode", editorContents.get("editor2").code);
+        model.addAttribute("submissionCodeStub", editorContents.get("editor1").code);
+        model.addAttribute("staticCode", editorContents.get("editor2").code);
         String implFileName = JavaClassGenerator.generateImplClassFilename(currentChallenge);
         String testFileName = JavaClassGenerator.generateTestClassFilename(currentChallenge);
-
-        model.addAttribute("implementationFileName", implFileName);
-        model.addAttribute("testFileName", testFileName);
-
+        
+        if (taskImplementation.getTask().getType() == ImplementationType.TEST) {
+            model.addAttribute("submissionTabFileName", testFileName);
+            model.addAttribute("staticTabFileName", implFileName);
+        } else {
+            model.addAttribute("submissionTabFileName", implFileName);
+            model.addAttribute("staticTabFileName", testFileName);
+        }
         return "task";
     }
 
     // TODO: This should actually be a separate service...
-    private TmcSubmission submitToTmc(Challenge challenge, String implementationCode, String testCode)
+    private TmcSubmission submitToTmc(Challenge challenge, String submissionCode,
+            String staticCode, ImplementationType taskType)
             throws IOException, ArchiveException {
         Map<String, byte[]> files = new HashMap<>();
-
         String implFileName = JavaClassGenerator.generateImplClassFilename(challenge);
         String testFileName = JavaClassGenerator.generateTestClassFilename(challenge);
 
-        files.put(implFileName, implementationCode.getBytes());
-        files.put(testFileName, testCode.getBytes());
-
+        if (taskType == ImplementationType.TEST) {
```

Comment:
This check could be moved out of submitToTmc, changing the parameters to just implCode and testCode.